llama_index/indices/vector_store/auto_retriever/prompts.py

```diff
@@ -0,0 +1,106 @@
+
+
+from typing import List
+
+from llama_index.indices.vector_store.auto_retriever.schema import (
+    MetadataInfo, VectorStoreInfo)
+from llama_index.prompts.base import Prompt
+from llama_index.prompts.prompt_type import PromptType
+from llama_index.vector_stores.types import (ExactMatchFilter,
+                                             VectorStoreQuerySpec)
+
+# single select
+PREFIX = """\
+Your goal is to structure the user's query to match the request schema provided below.
+
+<< Structured Request Schema >>
+When responding use a markdown code snippet with a JSON object formatted in the \
+following schema:
+
+{schema_str}
+
+The query string should contain only text that is expected to match the contents of \
```

Comment:
can the query be None so we only query by metadata filter?

Reply:
We can support this eventually. But I think currently our vector store implementation assumes query string is not None.